src/zenml/zen_stores/migrations/versions/8ed03137cacc_polymorthic_run_metadata.py

```diff
@@ -0,0 +1,159 @@
+"""polymorthic run_metadata [8ed03137cacc].
+
+Revision ID: 8ed03137cacc
+Revises: 14d687c8fa1c
+Create Date: 2023-11-17 16:19:34.281692
+
+"""
+import sqlalchemy as sa
+import sqlmodel
+from alembic import op
+from sqlmodel import Session
+
+# revision identifiers, used by Alembic.
+revision = "8ed03137cacc"
+down_revision = "14d687c8fa1c"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+    # ### commands auto generated by Alembic - please adjust! ###
+
+    with op.batch_alter_table("run_metadata", schema=None) as batch_op:
+        batch_op.add_column(
+            sa.Column("resource_type", sa.VARCHAR(length=255), nullable=True)
+        )
+        batch_op.add_column(
+            sa.Column(
+                "resource_id", sqlmodel.sql.sqltypes.GUID(), nullable=True
+            )
+        )
+        batch_op.drop_constraint(
+            "fk_run_metadata_step_run_id_step_run", type_="foreignkey"
+        )
+        batch_op.drop_constraint(
+            "fk_run_metadata_artifact_id_artifact", type_="foreignkey"
+        )
+        batch_op.drop_constraint(
+            "fk_run_metadata_pipeline_run_id_pipeline_run", type_="foreignkey"
+        )
+
+    with Session(bind=op.get_bind()) as session:
+        stmt = sa.text(
+            """
+            UPDATE run_metadata
+            SET resource_id = pipeline_run_id,
+                resource_type = 'pipeline run'
+            WHERE pipeline_run_id IS NOT NULL
+            """
+        )
+        session.execute(stmt)
+        stmt = sa.text(
+            """
+            UPDATE run_metadata
+            SET resource_id = step_run_id,
+                resource_type = 'step run'
```

Comment:
If you change enum naming convention, this would need to change too

Reply:
[b48e13f](https://github.com/zenml-io/zenml/pull/2064/commits/b48e13fc098f5f27cf68a5789079e6a9da69872a)